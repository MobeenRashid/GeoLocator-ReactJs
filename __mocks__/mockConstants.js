import { arrayToMap } from "../src/utils/arrayUtils";

const __mockLocationList = [
    {
        id: '1',
        address: 'Berlin',
        lng: "123",
        lat: "456"
    },
    {
        id: '2',
        address: 'Munich',
        lng: "124",
        lat: "457"
    }
];


const __mockState = {
    locations: __mockLocationList,
    requestingLocations: false
}

export { __mockState, __mockLocationList };