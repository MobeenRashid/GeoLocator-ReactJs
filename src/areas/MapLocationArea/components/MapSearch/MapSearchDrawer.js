import React from 'react';
import { connect } from 'react-redux';
import { Drawer } from 'antd';
import MapLocationSearch from './MapLocationSearch';
import BasicMap from '../../../../components/GoogleMap/BasicMap';
import DrawerFooter from './DrawerFooter';

const MapSearchDrawer = (props) => {
    return (
        <Drawer
            className="search-drawer"
            title="Search Location"
            placement="top"
            onClose={props.onClose}
            visible={props.visible}
            height={window.innerHeight}
        >
            <div className="drawer-content">
                <MapLocationSearch
                    disabled={props.disabled}
                    reset={!props.visible}
                    onLocationSelect={props.onLocationSelect}
                    defaultLocation={props.selectedLocation} />
                <BasicMap locations={props.locations} />
            </div>
            <DrawerFooter {...props}></DrawerFooter>
        </Drawer>
    );
}

function mapStateToProps(state) {
    return {
        mapKey: state.googleMapKey
    }
}

export default connect(mapStateToProps)(MapSearchDrawer);
