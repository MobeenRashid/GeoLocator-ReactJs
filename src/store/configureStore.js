import { createStore, applyMiddleware } from "redux";
import thunk from 'redux-thunk';

import appReducer from "../reducers/appReducer";

const configureStore = (initialState) => createStore(appReducer, initialState, applyMiddleware(thunk));
export default configureStore;