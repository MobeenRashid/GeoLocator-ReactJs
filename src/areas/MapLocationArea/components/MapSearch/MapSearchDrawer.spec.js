import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import MapSearchDrawer from './MapSearchDrawer';
import { Provider } from 'react-redux';
import configureStore from '../../../../store/configureStore';
import { __mockState } from '../../../../../__mocks__/__mockState';

const store = configureStore(__mockState);
describe("Map search drawer", () => {
    test("Should render correctly", () => {
        const component = mount(<Provider store={store}>
            <BrowserRouter>
                <MapSearchDrawer></MapSearchDrawer>
            </BrowserRouter>
        </Provider>);
        expect(component).toMatchSnapshot();
        component.unmount();
    });
});