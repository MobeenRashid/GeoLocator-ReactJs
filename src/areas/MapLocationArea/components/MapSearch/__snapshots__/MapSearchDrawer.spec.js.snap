// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map search drawer Should render correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(MapSearchDrawer)>
        <MapSearchDrawer
          dispatch={[Function]}
          mapKey=""
        >
          <Component
            className="search-drawer"
            height={768}
            placement="top"
            title="Search Location"
          >
            <Drawer
              className="search-drawer"
              closable={true}
              getPrefixCls={[Function]}
              height={768}
              level={null}
              mask={true}
              maskClosable={true}
              placement="top"
              prefixCls="ant-drawer"
              renderEmpty={[Function]}
              title="Search Location"
              width={256}
            >
              <Drawer
                className="search-drawer"
                closable={true}
                duration=".3s"
                ease="cubic-bezier(0.78, 0.14, 0.15, 0.86)"
                firstEnter={false}
                getContainer="body"
                getPrefixCls={[Function]}
                handler={false}
                height={768}
                level={null}
                mask={true}
                maskClosable={true}
                maskStyle={Object {}}
                onChange={[Function]}
                onHandleClick={[Function]}
                onMaskClick={[Function]}
                placement="top"
                prefixCls="ant-drawer"
                renderEmpty={[Function]}
                showMask={true}
                style={
                  Object {
                    "transform": undefined,
                    "zIndex": undefined,
                  }
                }
                title="Search Location"
                wrapperClassName=""
              />
            </Drawer>
          </Component>
        </MapSearchDrawer>
      </Connect(MapSearchDrawer)>
    </Router>
  </BrowserRouter>
</Provider>
`;
